code/src/compononets/TaskList.js

```diff
@@ -0,0 +1,33 @@
+import React from "react";
+import { useSelector, useDispatch } from "react-redux";
+
+import tasks from "../reducers/tasks";
+
+const TaskList = () => {
+  const items = useSelector((store) => store.tasks.items);
+  const dispatch = useDispatch();
+
+  return (
+    <div>
+      {items.map((task) => (
+        <div key={task.id} className="task-item">
+          <label className='task-text'>
+            <input
+              className='checkbox'
+              type="checkbox"
+              checked={task.complete}
+              onChange={() => dispatch(tasks.actions.toggleComplete(task.id))}
+            />
+            {task.text}
```

Comment:
You should be ready for accepting long strings which are not divided into the words- because your user can be not the best typer or just be a$#$@$% 😅 
![image](https://user-images.githubusercontent.com/50926159/116784814-864de400-aa96-11eb-9229-8755a16c0c28.png)

Reply:
Done